sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3038,14 +3046,20 @@ def _execute_task_n_nodes(self, handle: ResourceHandle, task: task_lib.Task,
         else:
             num_actual_nodes = task.num_nodes
 
+        cluster_external_ips = backend_utils.get_node_ips(handle.cluster_yaml,
```

Comment:
These two calls may take some time IIRC. Do we want to cache L3060 somewhere?

Reply:
One way would be to cache it in the `ResourceHandle` but I'm not sure if / how we would decide to invalidate the cache. Wouldn't it be possible that one of the workers dies and gets restarted? Is there a callback or path somewhere which would allow us to handle this case?

Reply:
Pushed a commit to handle this 👍.  We do need to invalidate the entry where appropriate.